### PR TITLE
h264encode : Added low_power command line option

### PR DESCRIPTION
**Pull request by ss22ongithub**
h264encode utility fails to run on the platforms where only low-power
(VAEntrypointEncSliceLP) entry points are supported . This change adds
a low_power command line option

### CODE REVIEW

File: encode/h264encode.c [hunk 1]
```diff
@@ -1088,17 +1104,29 @@ static int init_va(void)
         h264_profile = profile_list[i];
         vaQueryConfigEntrypoints(va_dpy, h264_profile, entrypoints, &num_entrypoints);
         for (slice_entrypoint = 0; slice_entrypoint < num_entrypoints; slice_entrypoint++) {
-            if (entrypoints[slice_entrypoint] == VAEntrypointEncSlice) {
+            if (requested_entrypoint == -1 ){
```
Comment (xhaihao) [inline encode/h264encode.c:1107]: Could you please use a space between  ')' and '{' ?

File: encode/h264encode.c [hunk 2]
```diff
@@ -1088,17 +1104,29 @@ static int init_va(void)
...
+                //Select the entry point based on what is avaiable
+                if ( (entrypoints[slice_entrypoint] == VAEntrypointEncSlice) ||
+                     (entrypoints[slice_entrypoint] == VAEntrypointEncSliceLP) ) {
+                    support_encode = 1;
+                    selected_entrypoint = entrypoints[slice_entrypoint];
+                    break;
+                }
+            }else if((entrypoints[slice_entrypoint] == requested_entrypoint)) {
```
Comment (xhaihao) [inline encode/h264encode.c:1115]: Add a space between '}' and 'else'

File: encode/h264encode.c [hunk 3]
```diff
@@ -1088,17 +1104,29 @@ static int init_va(void)
...
+                //Select the entry point based on what was requested in cmd line option 
                 support_encode = 1;
+                selected_entrypoint = entrypoints[slice_entrypoint];
                 break;
             }
         }
-        if (support_encode == 1)
+        if (support_encode == 1){
```
Comment (xhaihao) [inline encode/h264encode.c:1122]: Add a space between ')' and '{'.

File: encode/h264encode.c [hunk 4]
```diff
@@ -174,6 +174,11 @@ static unsigned int SyncPictureTicks=0;
 static unsigned int SavePictureTicks=0;
 static unsigned int TotalTicks=0;
 
+//Default entrypoint for Encode
+static VAEntrypoint requested_entrypoint = -1;
+static VAEntrypoint selected_entrypoint = -1;
+static int lp_option = 0;
```
Comment (xhaihao) [inline encode/h264encode.c:180]: lp_option is used in process_cmdline() only, so you needn't declare it here.

File: encode/h264encode.c (hunk 1 above)
Comment (ss22ongithub) [reply encode/h264encode.c]: done

File: encode/h264encode.c (hunk 2 above)
Comment (ss22ongithub) [reply encode/h264encode.c]: done

File: encode/h264encode.c (hunk 3 above)
Comment (ss22ongithub) [reply encode/h264encode.c]: done

File: encode/h264encode.c (hunk 4 above)
Comment (ss22ongithub) [reply encode/h264encode.c]: done

**Review by xhaihao — APPROVED** [review]
LGTM

**Pull request merged**